File: Python Exams/Basic Exam Collection/Basics Online Exam - 6 and 7 April 2019/06. Cinema Tickets/06. Cinema Tickets.py
move_name = input()
 
count_standard_tickets = 0
count_students_tickets = 0
count_kids_tickets = 0
 
while move_name != 'Finish':
    free_space = int(input())
    filled_space = 0
    while filled_space < free_space:
        ticket = input()
 
        if ticket == 'standard':
            filled_space += 1
            count_standard_tickets += 1
        elif ticket == 'student':
            filled_space += 1
            count_students_tickets += 1
        elif ticket == 'kid':
            filled_space += 1
            count_kids_tickets += 1
        elif ticket == 'End':
            break
 
    percent_full = filled_space * 100 / free_space
    print(f'{move_name} - {percent_full:.2f}% full.')
    move_name = input()
 
total_tickets = count_students_tickets + count_standard_tickets + count_kids_tickets
percent_students = count_students_tickets * 100 / total_tickets
percent_standard = count_standard_tickets * 100 / total_tickets
percent_kids = count_kids_tickets * 100 / total_tickets
 
print(f'Total tickets: {total_tickets}')
print(f'{percent_students:.2f}% student tickets.')
print(f'{percent_standard:.2f}% standard tickets.')
print(f'{percent_kids:.2f}% kids tickets.')

File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 6 and 7 April 2019/06. Cinema Tickets/06. Cinema Tickets.js
function cinemaTickets(arg) {
    let countStandardTickets = 0
    let countStudentsTickets = 0
    let countKidsTickets = 0
    let moveName = arg.shift()

    while (moveName != 'Finish') {
        let freeSpace = arg.shift()
        let filledSpace = 0

        while (filledSpace < freeSpace) {
            ticket = arg.shift()
            if (ticket === 'standard') {
                filledSpace++
                countStandardTickets++
            } else if (ticket == 'student') {
                filledSpace++
                countStudentsTickets++
            } else if (ticket == 'kid') {
                filledSpace++
                countKidsTickets++
            } else if (ticket === 'End')
                break
        }

        let percentFull = filledSpace * 100 / freeSpace
        console.log(`${moveName} - ${percentFull.toFixed(2)}% full.`)
        moveName = arg.shift()
    }

    let totalTickets = countStudentsTickets + countStandardTickets + countKidsTickets
    let percentStudents = countStudentsTickets * 100 / totalTickets
    let percentStandard = countStandardTickets * 100 / totalTickets
    let percentKids = countKidsTickets * 100 / totalTickets

    console.log(`Total tickets: ${totalTickets}`)
    console.log(`${percentStudents.toFixed(2)}% student tickets.`)
    console.log(`${percentStandard.toFixed(2)}% standard tickets.`)
    console.log(`${percentKids.toFixed(2)}% kids tickets.`)
}


// cinemaTickets([
// 'Taxi',
// '10',
// 'standard',
// 'kid',
// 'student',
// 'student',
// 'standard',
// 'standard',
// 'End',
// 'Scary Movie',
// '6',
// 'student',
// 'student',
// 'student',
// 'student',
// 'student',
// 'student',
// 'Finish'])